shipment_advice/models/shipment_advice.py

```diff
@@ -0,0 +1,385 @@
+        for shipment in self:
+            packages = self.loaded_move_line_ids.result_package_id
+            shipment.total_load = sum(packages.mapped("shipping_weight"))
+
+    @api.depends("planned_move_ids", "loaded_move_line_ids")
+    def _compute_picking_ids(self):
+        for shipment in self:
+            shipment.planned_picking_ids = shipment.planned_move_ids.picking_id
+            shipment.loaded_picking_ids = shipment.loaded_move_line_ids.picking_id
+
+    @api.depends("loaded_move_line_ids.package_level_id.package_id",)
+    def _compute_package_ids(self):
+        for shipment in self:
+            shipment.loaded_package_level_ids = (
+                shipment.loaded_move_line_ids.package_level_id
+            )
+            package_ids = set()
+            for line in shipment.loaded_move_line_ids:
+                if line.package_level_id:
+                    package_ids.add(line.package_level_id.package_id.id)
+            shipment.loaded_package_ids = self.env["stock.quant.package"].browse(
+                package_ids
+            )
+
+    @api.depends("planned_picking_ids", "planned_move_ids")
+    def _compute_count(self):
+        for shipment in self:
+            shipment.planned_pickings_count = len(self.planned_picking_ids)
+            shipment.planned_moves_count = len(self.planned_move_ids)
+            shipment.loaded_pickings_count = len(self.loaded_picking_ids)
+            shipment.loaded_move_lines_without_package_count = len(
+                self.loaded_move_line_without_package_ids
+            )
+            shipment.loaded_packages_count = len(self.loaded_package_ids)
+
+    @api.depends("planned_picking_ids", "loaded_picking_ids")
+    def _compute_carrier_ids(self):
+        for shipment in self:
+            if shipment.planned_picking_ids:
+                shipment.carrier_ids = shipment.planned_picking_ids.carrier_id
+            else:
+                shipment.carrier_ids = shipment.loaded_picking_ids.carrier_id
+
+    @api.model
+    def create(self, vals):
+        defaults = self.default_get(["name", "shipment_type"])
+        sequence = self.env.ref("shipment_advice.shipment_advice_outgoing_sequence")
+        if defaults["shipment_type"] == "incoming":
+            sequence = self.env.ref("shipment_advice.shipment_advice_incoming_sequence")
+        if vals.get("name", "/") == "/" and defaults.get("name", "/") == "/":
+            vals["name"] = sequence.next_by_id()
+        return super().create(vals)
+
+    def action_confirm(self):
+        for shipment in self:
+            if shipment.state != "draft":
+                raise UserError(
+                    _("Shipment {} is not draft, operation aborted.").format(
+                        shipment.name
+                    )
+                )
+            if not shipment.arrival_date:
+                raise UserError(
+                    _(
+                        "Arrival/departure date should be set on the "
```

Comment:
Is this exception description right, regarding the test just above ?
```suggestion
                        "Arrival date should be set on the "
```